web/src/message_events.js

```diff
@@ -466,24 +464,28 @@ export function update_messages(events) {
             }
         }
 
-        // Mark the message as edited for the UI. The rendering_only
-        // flag is used to indicated update_message events that are
-        // triggered by server latency optimizations, not user
-        // interactions; these should not generate edit history updates.
-        if (!event.rendering_only) {
-            msg.last_edit_timestamp = event.edit_timestamp;
-        }
+        if (msg !== undefined) {
+            // Mark the message as edited for the UI. The rendering_only
+            // flag is used to indicated update_message events that are
+            // triggered by server latency optimizations, not user
+            // interactions; these should not generate edit history updates.
+            if (!event.rendering_only) {
+                msg.last_edit_timestamp = event.edit_timestamp;
+            }
 
-        notifications.received_messages([msg]);
-        alert_words.process_message(msg);
+            notifications.received_messages([msg]);
+            alert_words.process_message(msg);
+        }
 
         if (topic_edited || stream_changed) {
             // if topic is changed
             let pre_edit_topic = util.get_edit_event_orig_topic(event);
             let post_edit_topic = new_topic;
 
             if (!topic_edited) {
-                pre_edit_topic = msg.topic;
+                if (msg !== undefined) {
+                    pre_edit_topic = msg.topic;
```

Comment:
I am not sure why we set `pre_edit_topic` is set again here if `!topic_edited`. The event includes the original topic when only stream is changed as well as per [the code here](https://github.com/zulip/zulip/blob/18d8fbc74be15ac20ee74433abe3263d8e8b7208/zerver/actions/message_edit.py#L505).